src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -846,110 +1021,103 @@ type TableContext<'TRecord> internal
+    /// NOTE: Throws <c>InvalidOperationException</c> if the table is not yet <c>Active</c>. It is recommended to ensure the Table is prepared via <c>VerifyTableAsync</c> or <c>VerifyOrCreateTableAsync</c> to guard against the potential for this state.
+    /// </summary>
+    /// <param name="throughput">Throughput configuration to use for the table. Always applied via either <c>CreateTable</c> or <c>UpdateTable</c>.</param>
+    /// <param name="streaming">Optional streaming configuration to apply for the table. Default (if creating): Disabled. Default: (if existing) do not change.</param>
+    /// <param name="custom">Callback to post-process the <c>UpdateTableRequest</c>. <c>UpdateTable</c> is inhibited if it returns <c>false</c> and no other configuration requires a change.</param>
+    /// <param name="currentTableDescription">Current table configuration, if known. Retrieved via <c>DescribeTable</c> if not supplied.</param>
+    member t.UpdateTableIfRequiredAsync(?throughput : Throughput, ?streaming, ?custom, ?currentTableDescription : TableDescription) : Async<unit> = async {
+        let! tableDescription = async {
+            match currentTableDescription with
+            | Some d -> return d
+            | None ->
+                match! Provisioning.tryDescribe (client, tableName) with
+                | Some d -> return d
+                | None -> return invalidOp "Table is not currently Active. Please use VerifyTableAsync or VerifyOrCreateTableAsync to guard against this state." }
+        match UpdateTableRequest.createIfRequired tableName tableDescription throughput streaming custom with
+        | None -> ()
+        | Some request -> do! UpdateTableRequest.execute client request }
 
-        do! verify None 10
-    }
+    /// <summary>
+    /// Asynchronously updates the underlying table with supplied configuration.<br/>
+    /// NOTE: The underlying API can throw if none the options represent a change or a change is in currently progress; see the DynamoDB <a href="https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateTable.html"><c>UpdateTable</c> API documentation</a>.
+    /// </summary>
+    /// <param name="throughput">Optional Throughput configuration to apply.</param>
+    /// <param name="streaming">Optional Streaming configuration to apply.</param>
+    /// <param name="customize">Callback to apply any further options desired.</param>
+    member _.UpdateTableAsync(?throughput, ?streaming, ?customize : UpdateTableRequest -> unit) : Async<unit> =
+        let request = UpdateTableRequest.create tableName
+        UpdateTableRequest.apply throughput streaming request
+        customize |> Option.iter (fun c -> c request)
+        UpdateTableRequest.execute client request
```

Comment:
Is there any value in retaining `UpdateTableAsync` - when would you want to use this over IfRequired?

Reply:
its used in the impl of the (obsoleted) UpdateProvisionedThroughputAsync and UpdateProvisionedThroughput so I could only mark it `internal`

Unless you want me to whack those two also ? (I agree it it kinda doesnt sense to not arbitrarily wrap things given e.g. DescribeTable is not wrapped)

Reply:
I think maybe make this one internal, it would be better to not add a public method we know we want to get rid of